runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -252,20 +256,78 @@ ROMClassBuilder::buildROMClass(ROMClassCreationContext *context)
 }
 
 BuildResult
-ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda)
+ROMClassBuilder::injectIdentityObjectInterface(J9CfrClassFile *classfile)
+{
+		BuildResult result = OK;
+		PORT_ACCESS_FROM_PORT(_portLibrary);
+		UDATA nameLen = strlen(IDENTITY_OBJECT);
+		_identityObjectNameBuffer = (U_8*)j9mem_allocate_memory(nameLen, J9MEM_CATEGORY_CLASSES);
+		if (NULL == _identityObjectNameBuffer) {
+			result = OutOfMemory;
+		} else {
+			/* add utf8 slot */
+			memcpy(_identityObjectNameBuffer, IDENTITY_OBJECT, nameLen);
+			U_16 utf8SlotIndex = classfile->constantPoolCount;
+			J9CfrConstantPoolInfo* utf8Slot = &(classfile->constantPool[utf8SlotIndex]);
+			utf8Slot->bytes = _identityObjectNameBuffer;
+			utf8Slot->slot1 = nameLen;
+			utf8Slot->slot2 = 0;
+			utf8Slot->tag = CFR_CONSTANT_Utf8;
+			utf8Slot->flags1 = 0;
+			utf8Slot->nextCPIndex = 0;
+			utf8Slot->romAddress = 0;
+
+			if (0 == classfile->firstUTF8CPIndex) {
+				classfile->firstUTF8CPIndex = utf8SlotIndex;
+			} else {
+				classfile->constantPool[classfile->lastUTF8CPIndex].nextCPIndex = utf8SlotIndex;
+			}
+			classfile->lastUTF8CPIndex = utf8SlotIndex;
+
+			classfile->constantPoolCount += 1;
+
+			/* add class slot */
+			U_16 classSlotIndex = classfile->constantPoolCount;
+			J9CfrConstantPoolInfo* classSlot = &(classfile->constantPool[classSlotIndex]);
+			classSlot->bytes = _identityObjectNameBuffer;
+			classSlot->slot1 = utf8SlotIndex;
+			classSlot->slot2 = 0;
+			classSlot->tag = CFR_CONSTANT_Class;
+			classSlot->flags1 = 0;
+			classSlot->nextCPIndex = 0;
+			classSlot->romAddress = 0;
+
+			classfile->constantPoolCount += 1;
+
+			classfile->interfaces[classfile->interfacesCount] = classSlotIndex;
+			classfile->interfacesCount++;
+		}
+
+		return result;
+}
+
+
+
+BuildResult
+ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda, bool dontUselastCPEntry)
 {
 	J9CfrConstantPoolInfo* constantPool = classfile->constantPool;
 	U_32 cpThisClassUTF8Slot = constantPool[classfile->thisClass].slot1;
 	U_32 originalStringLength = constantPool[cpThisClassUTF8Slot].slot1;
 	const char* originalStringBytes = (const char*)constantPool[cpThisClassUTF8Slot].bytes;
-	U_16 newUtfCPEntry = classfile->constantPoolCount - 1; /* last cpEntry is reserved for anonClassName utf */
+	U_16 newUtfCPEntry = classfile->constantPoolCount; /* last cpEntry is reserved for anonClassName utf */
 	U_32 i = 0;
 	BOOLEAN stringOrNASReferenceToClassName = FALSE;
 	BOOLEAN newCPEntry = TRUE;
 	UDATA newAnonClassNameLength = originalStringLength + 1 + ROM_ADDRESS_LENGTH + 1;
 	BuildResult result = OK;
 	PORT_ACCESS_FROM_PORT(_portLibrary);
 
+	if (dontUselastCPEntry) {
+		/* two extra slots where added for the IdentityObject interface */
+		newUtfCPEntry -= 2;
```

Comment:
This is actually not needed, I forgot about this when I changed the way CP entries are added for anonClassNames. This is a bug.

Reply:
The `dontUselastCPEntry` param can go away